internal/matrix/matrix.go

```diff
@@ -1,33 +1,26 @@
-package helper
+package matrix
 
 import (
 	"errors"
 )
 
-func ClientErrorHandling (rows int, columns int) error {
+func GenerateFibMatrix(rows int, columns int) ([][]int64, error) {
 
 	//the max value for a int64 is 9,223,372,036,854,770,000 
 	//the largest fibonacci number int64 can handle is the 92nd number
 	//this function puts in constraints so that the server never encounters a Fibonacci number that's too large to handle
 	if rows < 1 || rows >= 92 || columns < 1 || columns >= 92 {
-		return errors.New("You must enter a number greater than 0 and less than 93")
+		return nil, errors.New("You must enter a number greater than 0 and less than 93")
 	}
 	
 	if rows*columns > 92 {
-		return errors.New("Can only support a matrix with maximum of 92 numbers")
+		return nil, errors.New("Can only support a matrix with maximum of 92 numbers")
```

Comment:
It might be a good idea to declare these errors as variables, e.g. `dimensionError` and `dimensionOutOfBoundsError` just to say something. The problem here is that if you ever change the wording of your error message, your tests will break unnecessarily. There's more advance techniques you could use here like declaring a custom error type and testing against type instead of value, but that's a bit to overkill for this. You could also use simpler techniques such as only testing for error existence and not equality to avoid the false positive in the feature. I'll leave the decision to you. We can talk about this in more detail if you'd  like.

Reply:
Thats a great callout. I did change the wording and did unnecessarily break my tests so have already encountered this problem and would like to avoid it again in the future! Will look into the different approaches now

Reply:
I went with checking for the existence of an error, but I'd be interested in talking through how declaring errors as variables would work. Ill add it to our 1x1 agenda for tomorrow